### PR TITLE
cnap/core: add the implementation for frame class

### LINKED ISSUE #34: design frame class for transferring data between micro services

### PR DESCRIPTION
**Pull request by Hulongyin**
This PR is to fix #34 , design and implement frame class for transferring data between micro services.

### CODE REVIEW

File: cnap/core/frame.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+LOG = logging.getLogger(__name__)
+
+# pylint: disable=no-member
+
+class FrameCipherBase(ABC):
+    """
+    Abstract base class frame cipher implementations.
+    """
+
+    def __init__(self):
```
Comment (leyao-daily) [inline cnap/core/frame.py]: remove the __init__ method if it's not doing anything

File: cnap/core/frame.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        return self._ts_infer_start
+
+    @timestamp_infer_start.setter
+    def timestamp_infer_start(self, timestamp: float) -> None:
+        """
+        Set the timestamp of the inference's start point.
+        """
+        self._ts_infer_start = timestamp
+
+    def to_blob(self) -> Any:
```
Comment (leyao-daily) [inline cnap/core/frame.py]: Maybe we can change to more specified hint instead of `Any`

File: cnap/core/frame.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        frame_msg.ts_infer_start = self._ts_infer_start
+
+        serialized_data = frame_msg.SerializeToString()
+        # TODO: zlib, gzip or QATZip to accelerate the compression
+        compressed_data = serialized_data
+
+        return compressed_data
+
+    @staticmethod
+    def from_blob(blob: Any) -> Any:
```
Comment (leyao-daily) [inline cnap/core/frame.py]: Same.

File: cnap/core/frame.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        """
+        Restore a frame instance from a blob.
+
+        Returns:
+            Any: the frame restored from a blob.
+        """
+        # TODO: zlib, gzip or QATZip to accelerate the decompression
+        decompressed_data = blob
```
Comment (leyao-daily) [inline cnap/core/frame.py]: For protobuf serializing, we may add `try catch` to raise error.

File: cnap/core/frame.py (hunk 1 above)
Comment (Hulongyin) [reply cnap/core/frame.py]: Thanks, fixed.

File: cnap/core/frame.py (hunk 2 above)
Comment (Hulongyin) [reply cnap/core/frame.py]: Thanks, fixed.

File: cnap/core/frame.py (hunk 3 above)
Comment (Hulongyin) [reply cnap/core/frame.py]: Thanks, fixed.

File: cnap/core/frame.py (hunk 4 above)
Comment (Hulongyin) [reply cnap/core/frame.py]: Thanks, fixed.

**Review by leyao-daily — APPROVED** [review]
LGTM.

**Pull request merged**